pkg/core/requestmatcher.go

```diff
@@ -50,20 +50,29 @@ func (request *RequestMatcher) Matches(x interface{}) bool {
 	request.target = target
 
 	match := request.request.Method == target.Method && (request.request.URL.Path == target.URL.Path ||
-		request.request.URL.Path == target.URL.Opaque)
+		request.request.URL.Opaque == target.URL.Opaque)
 
 	if match {
 		match = matchHeader(request.request.Header, request.target.Header)
+		if !match {
+			Logger.Debug(fmt.Sprintf("got header: %v, want header %v", request.target.Header, request.request.Header))
```

Comment:
I don't recommend using `Logger` here. This source file does not rely on any other source files. You just broke it.

Please consider finding a better way to print the debug info if you think it's necessary. For example, you can print the additional information [here](https://github.com/jenkins-zh/jenkins-client/pull/14/files#diff-394bec2992960292436808ad64323b1f0b849f03758e25e922ee57a69ca20f31R114).

Reply:
Please confirm that the `here` link is well, thx.

Reply:
By default, debug level won't be printed out. If someone encountered something wrong with test, he/her can enable debug log level and found what was different between got and want requests.

Reply:
Okay, I'll try.

Reply:
> By default, debug level won't be printed out. If someone encountered something wrong with test, he/her can enable debug log level and found what was different between got and want requests.

So, who can know how to set the logger level if they met a problem.

No matter what, adding a `Logger` will make this source file complicated.